doc/adr/0005-i18n-with-dynamic-string.md

```diff
@@ -0,0 +1,50 @@
+# 5. i18n with dynamic string
+
+Date: 2020-11-10
+
+## Status
+
+Accepted
+
+## Context
+
+We need to provide i18n string with some requirements:
+
+- dynamic string
+
+- support plurals
+
+## Decision
+
+Need to implement it with `intl` 
+
+## Consequences
+
+Set up:
+
+1. Add localization dependencies
+```
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_localizations: # add this
+    sdk: flutter # add this
+# other lines
+dev_dependencies:  
+  intl_translation: {lastest version}
+```
+
+2. Localization class
+
+3. The `.arb` (Application Resource Bundle) file
+```
+flutter pub pub run intl_translation:extract_to_arb --output-dir=lib/l10n lib/localizations.dart
```

Comment:
I don't understand this command, we have to compile `localizations.dart` by ourselves ? We don't push it ?

Reply:
please read this article: https://proandroiddev.com/flutter-localization-step-by-step-30f95d06018d

Reply:
nice, can you also add the article to this `adr`

Reply:
I summarize it as short as possible in the adr. I concern that article is no longer on the internet

Reply:
Thanks for the link